### PR TITLE
[CTP] include BC shifts for 2023

### PR DESCRIPTION
**Pull request by mfasDa**
Separate cases 2022 with 294 BC shifts for
certain periods and 2023 with 3 BC shifts for
certain periods

### CODE REVIEW

**Comment by mfasDa** [comment]
@ekryshen Please review!

The 2022 case is still not complete since for LHC22s (PbPb) the shift in the CTP digits is not 294 but 293 BCs.

File: Common/TableProducer/eventSelection.cxx [hunk 1]
```diff
@@ -206,7 +206,18 @@ struct BcSelectionTask {
...
       int run = bcs.iteratorAt(0).runNumber();
-      triggerBcShift = (run <= 526766 || (run >= 526886 && run <= 527237) || (run >= 527259 && run <= 527518) || run == 527523 || run == 527734 || run >= 534091) ? 0 : 294;
+      // Corection for BC shifts in CTP wrt LM
+      if (run >= 516967 && run <= 530305) {
+        // 2022
+        // all runs with CTP readout need to be corrected for L1L0+L0LM delay (294 BCs), runs with replayed readout (LHC22m and part of LHC22o) have the correct BC
+        triggerBcShift = (run <= 526766 || (run >= 526886 && run <= 527237) || (run >= 527259 && run <= 527518) || run == 527523 || run == 527734) ? 0 : 294;
+      } else if (run >= 530306 && run <= 545563) {
+        // 2023
+        // all runs from periods LHC23zd - LHC23zs have a 2 BC shift in CTP which needs to be corrected
```
Comment (ekryshen) [inline Common/TableProducer/eventSelection.cxx:216]: It would be good to adjust the comment (3 BC shift) to avoid confusion

**Comment by ekryshen** [comment]
Hi @mfasDa, the changes are fine with me.

**Comment by mfasDa** [comment]
Please keep the PR pending - since the apass1 is going to be repeated - with now fixed scripts in O2DPG - it should hopefully be fixed in the reconstruction by then

**Comment by mfasDa** [comment]
I have checked example runs from LHC23zl and LHC23zs and see the BC distribution now aligned in the new apass1. Consequently shifts for 2023 must no longer be applied. Consequently the PR is closed.

**Pull request closed without merging**